seonjeong/app.js

```diff
@@ -51,6 +51,145 @@ app.post("/users", async (req, res) => {
+
+  await appDataSource.query(
+    `INSERT INTO posts (
+      title,
+      content,
+      image_url,
+      user_id
+    ) VALUES (?, ?, ?, ?);`,
+    [title, content, imageUrl, userId]
+  );
+  return res.status(201).json({ message: "postCreated" });
+});
+//동기: 융통성x, 비동기: 융통성o
+//async + await
+
+app.get("/search", async (req, res) => {
+  const result = await appDataSource.query(
+    `SELECT
+    users.id as userId,
+    users.profile_image as userProfileImage,
+    posts.id as postingId,
+    posts.image_url as postingImageUrl,
+    posts.content as postingContent
+    FROM users INNER JOIN posts
+    ON users.id = posts.user_id
+    `
+  );
+
+  return res.status(200).json({ data: result });
+});
+
+app.get("users/userId/posts", async (req, res) => {
+  const { userId } = req.params;
+
+  const result = await appDataSource.query(
+    `
+  SELECT
+    u.id as userId,
+    u.profile_image as userProfileImage,
+    JSON_ARRAYAGG(
+      JSON_OBJECT(
+        "postingId", p.id,
+        "postingImageUrl", p.image_url,
+        "postingTitle", p.title,
+        "postingContent", p.content
+      )
+    ) as postings
+    FROM users as u
+    JOIN posts as p ON p.user_id = u.id
+    WHERE u.id = ?
+    GROUP BY u.id
+    `,
+    [userId]
+  );
+
+  return res.status(200).json({ data: result });
+});
+//ARRAYAGG: 객체를 배열로 묶어주는 역할
+//GROUP BY: 특정 컬럼을 그룹화함
+
+app.patch("/posts/postId", async (req, res) => {
+  const { postId } = req.params;
+  const { content } = req.body;
+
+  await appDataSource.query(
+    `
+    UPDATE posts
+    SET content = ?
+    WHERE id = ?
+  `,
+    [content, postId]
+  );
+
+  const result = await appDataSource.query(
+    `
+    SELECT
+      u.id as userId,
+      u.name as userName,
+      p.id as postingId,
+      p.title as postingTitle,
+      p.content as postingContent
+    FROM posts p
+    JOIN users u ON u.id = p.user_id
+    WHERE p.id = ?
+  `,
+    [postId]
+  );
+
+  return res.status(201).json({ message: "postModify" });
+});
+//patch: 리소스의 부분적인 수정을 할 때에 사용
+
+app.delete("/posts/postId", async (req, res) => {
+  const { postId } = req.params;
+
+  await appDataSource.query(
+    `
+  DELETE FROM posts
+  WHERE id = ?
+  `,
+    [postId]
+  );
+
+  return res.status(200).json({ message: "postDeleted" });
+});
+
+app.post("/likes", async (req, res) => {
+  const { userId, postId } = req.body;
+
+  const [isLiked] = await appDataSource.query(
+    `SELECT id
+    FROM likes
+    WHERE user_id = ?
+    AND post_id = ?`
+  );
+
+  if (!isLiked) {
+    await appDataSource.query(
+      `INSERT INTO likes (
+        user_id, post_id
+      ) VALUES (?, ?)`,
+      [userId, postId]
+    );
+
+    return res.status(201).json({ message: "likeCreatedD" });
+  } else {
+    await appDataSource.query(
+      `DELETE FROM likes
+      WHERE user_id = ?
+      AND post_id = ?`,
+      [userId, postId]
+    );
+
+    return res.status(201).json({ message: "likeDeleted" });
```

Comment:
삭제 됐을 경우에는 status code를 201이 아닌, 200이나 204를 사용해 주세요.
```suggestion
    return res.status(200).json({ message: "likeDeleted" });
```